prow/cmd/deck/jobs.go

```diff
@@ -134,6 +137,25 @@ func (ja *JobAgent) GetJobLog(job, id string) ([]byte, error) {
 	return nil, fmt.Errorf("cannot get logs for prowjob %q with agent %q: the agent is missing from the prow config file", j.Metadata.Name, j.Spec.Agent)
 }
 
+func labelsMatch(l1, l2 map[string]string) bool {
```

Comment:
This function checks that the maps are equal which is not what it should be checking. Instead is should check that the required labels are a subset of the labels on the job.